services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProvider.java

```diff
@@ -157,11 +157,15 @@ public Response performLogin(AuthenticationRequest request) {
             Boolean allowCreate = null;
             if (getConfig().getConfig().get(SAMLIdentityProviderConfig.ALLOW_CREATE) == null || getConfig().isAllowCreate())
                 allowCreate = Boolean.TRUE;
+            Boolean forceAuthn = getConfig().isForceAuthn();
+            if ("login".equals(request.getAuthenticationSession().getClientNote(OIDCLoginProtocol.PROMPT_PARAM)) ||
```

Comment:
I would slightly prefer something like:
```
LoginProtocol protocol = context.getSession().getProvider(LoginProtocol.class, request.getAuthenticationSession.getProtocol());
if (protocol.requireReauthentication(null, request.getAuthenticationSession()) 
```
It might need some small updates to the `OIDCLoginProtocol.requireReauthentication` to allow null userSession to be passed as a parameter.

@rmartinc @phamann WDYT?

Reply:
That is certainly cleaner. I'm happy to give it a go 👍🏻

Reply:
@mposolda It's more generic so I think that it's better too. @phamann As Marek comments you need to also modify the [OIDCLoginProtocol](https://github.com/keycloak/keycloak/blob/main/services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocol.java#L414) to allow passing a null user session (return false in that case). The other protocols does not use the user session, so OIDC is the only one affected.